File: vanillapull/webapp/src/main/java/com/octo/vanillapull/service/scala/MockMonteCarlo.scala
package com.octo.vanillapull.service.scala

import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Service

@Profile(Array("mock"))
@Service
class MockMonteCarlo extends PricingService {

   def calculatePrice(maturity: Double, spot: Double, strike: Double, volatility: Double): Double = {
     return scala.math.random
   }
 }

File: vanillapull/webapp/src/main/java/com/octo/vanillapull/service/synchronization/MockMonteCarlo.java
package com.octo.vanillapull.service.synchronization;

import com.octo.vanillapull.service.PricingService;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;

import java.util.concurrent.ThreadLocalRandom;

/**
 * Created by Bastien on 16/04/2014.
 */
@Profile("mock")
@Service
public class MockMonteCarlo implements PricingService {

    @Override
    public double calculatePrice(double maturity, double spot, double strike, double volatility) {
        return ThreadLocalRandom.current().nextDouble();
    }

    @Override
    public void init() throws Exception {

    }
}
